src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -449,6 +452,20 @@ private void processIter(DetailAST root, AstState astState) {
         }
     }
 
+    /**
+     * Creates a new {@link SortedSet} with a deterministic order based on the
+     * Check's name before the default ordering.
+     *
+     * @return The new {@link SortedSet}.
+     */
+    private static SortedSet<AbstractCheck> createNewCheckSortedSet() {
+        return new TreeSet<>(
+                Comparator.<AbstractCheck, String>comparing(check -> check.getClass().getName())
+                        .thenComparing(Comparator.comparing(AbstractCheck::getId,
+                                Comparator.nullsLast(Comparator.naturalOrder())))
+                        .thenComparing(AbstractCheck::hashCode));
```

Comment:
can you remind me reason of this sorting ?

each JVM execution it very likely be different. In the same JVM execution we order Checks ones, so order of them will be same during execution.

Reply:
Reason for `hashCode` sorting is at https://github.com/checkstyle/checkstyle/pull/9254#discussion_r571646551